build.gradle

```diff
@@ -95,29 +95,38 @@ subprojects {
   dependencies {
     compile "org.apache.logging.log4j:log4j-core:$log4jVersion"
     runtime "org.apache.logging.log4j:log4j-slf4j-impl:$log4jVersion"
-    testCompile project(path: ':', configuration: 'testArchives') 
+    testCompile project(path: ':', configuration: 'testArchives')
   }
-  
+
   task cleanServer << {
     delete 'locator'
     delete 'server1'
     delete 'server2'
   }
   clean.finalizedBy cleanServer
 
+  task replaceVersion(type:Copy) {
+    from "$projectDir/scripts/start.gfsh"
+    from "$projectDir/scripts/stop.gfsh"
+    into (project.getBuildDir())
+    expand(version: "${version}")
+  }
+  compileJava.dependsOn replaceVersion
```

Comment:
Why is this needed?

Reply:
Prior to this change, someone needed to manually synchronize the version string in `gradle.properties` with `scripts/start.gfsh` so `gfsh` would correctly `deploy` the JAR file. This automates that synchronization by making copies of `*.gfsh` with `${version}` replaced with the version number found in `gradle.properties`.

Reply:
Ah, I get it.  I'd like to keep a consistent pattern across all examples.  So we should either update the other example README's to use `build` instead of `scripts`, or remove `${version}` from the `function` jar filename.  What do you think?

Reply:
I'm in favor of uniformity. I didn't do it as this PR because I didn't know if there was consensus and worried that it was scope creep for GEODE-3440.

Reply:
I'd like to settle on an approach before merging this PR.  I'm leaning towards removing the version from the jar filename.  Thoughts?  I'm happy to assist with gradle-y stuff.

Reply:
I would have preferred removing the version number from the JAR file in the first place but water cooler scuttlebutt was that folks _really_ wanted to have everything versioned.

Reply:
Ideally we should be creating a separate jar containing only the code that needs to be deployed on the server*.  I like the approach of using `gfsh deploy` (because that's the user pattern we want to encourage).  We just need to enhance the build template to create a deployable jar.

* The `loader` example gets around this by starting the server with a classpath containing the build tree.  It would be better to NOT do this.